cmd/bnsd/x/blueaccount/codec.proto

```diff
@@ -0,0 +1,192 @@
+syntax = "proto3";
+
+package blueaccount;
+
+import "codec.proto";
+import "gogoproto/gogo.proto";
+
+// Domain represents a blue account domain.
+//
+// Domain is a container for accounts. Account representation is
+// `<name>*<domain>`.
+// Each domain owner controls all accounts that belong to that domain.
+// Each domain controls expiration of all accounts that belong to that domain.
+//
+// Because of the functionality that the Red Account provides, Blue Account
+// domain must not be `iov` as this is a domain used by Red Account
+// functionality.
+message Domain {
+  weave.Metadata metadata = 1;
+  string domain = 2;
+  // Owner is a weave.Address that controls this domain and all accounts that
+  // belong to this domain.
+  bytes owner = 3 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
+  // Valid till defines the expiration date for the domain. Expired domain
+  // cannot be used or modified. None of the accounts that belong to an expired
+  // domain can be used of modified as well.
+  int64 valid_till = 4 [(gogoproto.casttype) = "github.com/iov-one/weave.UnixTime"];
+}
+
+// Account represents a blue account name.
+//
+// String representation of an account is `<name>*<domain>`. Empty name is an
+// acceptable account string. Complete validation rules for name and domain are
+// declared by configuration entity.
+//
+// An account must always belong to an existing domain.
+message Account {
+  weave.Metadata metadata = 1;
+  // Domain references a domain that this account belongs to.
+  string domain = 2;
+  string name = 3;
+  // Owner is a weave.Address that controls this account. Can be empty.
+  //
+  // An account is always owned by the domain owner and as such, regardless
+  // the owner field value, can be edited by the domain owner that this account
+  // belongs to.
```

Comment:
```
An account is always owned by the domain owner and as such, regardless
  // the owner field value, can be edited by the domain owner that this account
  // belongs to.
```
To be honest I find this comment difficult to understand.

Reply:
How about:

Reply:
Much better